test/functional/apps/discover/group2/_adhoc_data_views.ts

```diff
@@ -0,0 +1,127 @@
+
+      await PageObjects.timePicker.setDefaultAbsoluteRangeViaUiSettings();
+      await PageObjects.common.navigateToApp('discover');
+    });
+
+    it('should navigate back correctly from to surrounding and single views', async () => {
+      await PageObjects.discover.createAdHocDataView('logstash', true);
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      // navigate to context view
+      await dataGrid.clickRowToggle({ rowIndex: 0 });
+      const [, surrDocs] = await dataGrid.getRowActions({ rowIndex: 0 });
+      await surrDocs.click();
+      await PageObjects.context.waitUntilContextLoadingHasFinished();
+
+      await find.clickByCssSelector(`[data-test-subj="breadcrumb first"]`);
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      expect(await PageObjects.discover.getCurrentlySelectedDataView()).to.be('logstash*');
+
+      // navigate to single doc view
+      await dataGrid.clickRowToggle({ rowIndex: 0 });
+      const [singleView] = await dataGrid.getRowActions({ rowIndex: 0 });
+      await singleView.click();
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      await find.clickByCssSelector(`[data-test-subj="breadcrumb first"]`);
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      expect(await PageObjects.discover.getCurrentlySelectedDataView()).to.be('logstash*');
+    });
+
+    it('should support query and filtering', async () => {
+      await filterBar.addFilter('nestedField.child', 'is', 'nestedValue');
+      expect(await filterBar.hasFilter('nestedField.child', 'nestedValue')).to.be(true);
+      await retry.try(async function () {
+        expect(await PageObjects.discover.getHitCount()).to.be('1');
+      });
+      await filterBar.removeFilter('nestedField.child');
+
+      await queryBar.setQuery('test');
+      await queryBar.submitQuery();
+      await retry.try(async () => expect(await PageObjects.discover.getHitCount()).to.be('22'));
+
+      await queryBar.clearQuery();
+      await queryBar.submitQuery();
+    });
+
+    it('should not update data view id when saving search first time', async () => {
+      const prevDataViewId = await getCurrentDataViewId();
+
+      await PageObjects.discover.saveSearch('hoc-data-view-search');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      const newDataViewId = await getCurrentDataViewId();
+
+      expect(prevDataViewId).to.equal(newDataViewId);
+    });
+
+    it('should update data view id when saving new search copy', async () => {
+      const prevDataViewId = await getCurrentDataViewId();
+
+      await PageObjects.discover.saveSearch('hoc-data-view-search-new', true);
+      await PageObjects.header.waitUntilLoadingHasFinished();
+
+      const newDataViewId = await getCurrentDataViewId();
+
+      expect(prevDataViewId).not.to.equal(newDataViewId);
+    });
+
+    it('should update data view id when saving data view from hoc one', async () => {
```

Comment:
Thx a lot for this function, mind if I ask for another one? I think we should at alt least one test to add a saved search with an ad hoc one to a dashboard 🙏

Reply:
Test with saved search embeddables added.